### ISSUE TITLE
ICE with `const_evaluatable_checked`

### ISSUE THREAD

**Issue by lukaslueg** [rust-lang/rust#82418]
The following code will cause an ICE [(playground)](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e22ed5a309e2a886c3922e6f5094eedb) with `const_evaluatable_checked`. There are warnings that crashes are about to be expected, I'm reporting nevertheless.

cc #44580, #76560

```rust
#![feature(const_generics)]
#![feature(const_evaluatable_checked)]

pub struct Foobar<T, const N: usize> {
    t: T,
}

pub trait Footrait {
    const N: usize;
}

impl<T> Footrait for T {
    const N: usize = 0;
}

pub fn new_foo<T>(t: T) -> Foobar<T, { <T as Footrait>::N }>
where
    T: Footrait,
{
    Foobar { t }
}

fn main() {
    let foo = new_foo(0);
}
```

`rustc --version --verbose`:
```
rustc 1.52.0-nightly (3e826bb11 2021-02-21)
```

### Error output
```
query stack during panic:
#0 [codegen_fulfill_obligation] checking if `Footrait` fulfills its obligations
thread 'rustc' panicked at 'substs of instance DefId(0:8 ~ playground[3b56]::Footrait::N) not normalized for codegen: [^0]', compiler/rustc_middle/src/ty/instance.rs:284:9
...
```

**Comment by lcnr** [comment]
same underlying issue as #83249, closing in favor of that one
